static/js/Autocomplete.jsx

```diff
@@ -0,0 +1,489 @@
+import Sefaria from "./sefaria/sefaria";
+import React, {useEffect, useState} from "react";
+import classNames from "classnames";
+import {EnglishText, HebrewText, InterfaceText, SearchButton} from "./Misc";
+import { useCombobox } from 'downshift';
+
+const type_icon_map = {
+  "Collection": "collection.svg",
+  "AuthorTopic": "iconmonstr-pen-17.svg",
+  "TocCategory": "iconmonstr-view-6.svg",
+  "PersonTopic": "iconmonstr-hashtag-1.svg",
+  "Topic": "iconmonstr-hashtag-1.svg",
+  "ref": "iconmonstr-book-15.svg",
+  "search": "iconmonstr-magnifier-2.svg",
+  "Term": "iconmonstr-script-2.svg",
+  "User": "profile.svg"
+};
+
+const type_title_map = {
+  "Collection": "Collections",
+  "AuthorTopic": "Authors",
+  "TocCategory": "Categories",
+  "PersonTopic": "Topics",
+  "Topic": "Topics",
+  "ref": "Books",
+  "search": "",
+  "Term": "Terms",
+  "User": "Users"
+};
+
+const type_icon = function(itemType, itemPic) {
```

Comment:
Use the syntax `function type_icon(itemType, itemPic)` which is used directly below.